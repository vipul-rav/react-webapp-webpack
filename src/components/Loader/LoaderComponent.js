import React, { memo } from 'react';
import PropTypes from 'prop-types';

const LoaderComponent = memo(({ content, isLoading = false }) =>
  isLoading ? (
    <div id="preloader">
      <div id="loader">
        {content.loadingText}
        <div className="spin" />
      </div>
    </div>
  ) : null
);

LoaderComponent.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  content: PropTypes.object.isRequired
};

LoaderComponent.defaultProps = {
  content: {
    loadingText: 'Please wait....'
  }
};

export { LoaderComponent };
