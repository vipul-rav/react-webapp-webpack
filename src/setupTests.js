import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

let store;
beforeEach(() => {
  const mockStore = configureMockStore();
  store = mockStore();
});

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
