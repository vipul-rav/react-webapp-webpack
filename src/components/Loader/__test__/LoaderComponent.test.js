import React from 'react';
import { render } from '../../../setupTests';
import { LoaderComponent } from '../LoaderComponent';
import configureMockStore from 'redux-mock-store';

const content = {
  ExitText: 'Exit',
  btnContinue: 'Continue',
  homePageText: 'Home Page'
};

const setup = (overridesProps) => {
  const props = {
    content: content,

    ...overridesProps
  };

  const mockStore = configureMockStore();
  const store = mockStore({
    config: {
      externalContent: content
    }
  });

  const { container, getByText } = render(<LoaderComponent {...props} store={store} />);

  return {
    props,
    container,
    getByText
  };
};

describe('Loader Component', () => {
  it('should render isLoading=true', () => {
    const overrideProps = {
      isLoading: true,
      content: {
        loadingText: 'Please wait...'
      }
    };
    const { container } = setup(overrideProps);
    expect(container).toMatchSnapshot();
  });
  it('should render isLoading=false', () => {
    const overrideProps = {
      isLoading: false,
      content: {
        loadingText: 'Please wait...'
      }
    };
    const { container } = setup(overrideProps);
    expect(container).toMatchSnapshot();
  });
});
