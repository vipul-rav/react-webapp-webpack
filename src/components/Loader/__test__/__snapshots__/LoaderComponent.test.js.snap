// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component should render isLoading=false 1`] = `<div />`;

exports[`Loader Component should render isLoading=true 1`] = `
<div>
  <div
    id="preloader"
  >
    <div
      id="loader"
    >
      Please wait...
      <div
        class="spin"
      />
    </div>
  </div>
</div>
`;
